contracts/axelar-gateway/src/testutils.rs

```diff
@@ -87,13 +98,42 @@ pub fn generate_signers_set(
     env: &Env,
     num_signers: u32,
     domain_separator: BytesN<32>,
+) -> TestSignerSet {
+    generate_signers_set_with_randomness(env, num_signers, domain_separator, true)
+}
+
+pub fn generate_deterministic_signers_set(
+    env: &Env,
+    num_signers: u32,
+    domain_separator: BytesN<32>,
+) -> TestSignerSet {
+    generate_signers_set_with_randomness(env, num_signers, domain_separator, false)
+}
+
+fn generate_signers_set_with_randomness(
```

Comment:
Instead of adding more complexity to helpers, can you try seeding the rng with a deterministic string at the beginning of the test that needs to be deterministic? The rng will then be deterministic and so none of the helpers need to change